### PR TITLE
added support for since argument for fwdingHistory

### PR DESCRIPTION
**Pull request by jrmillerwork**
This is related to issue #1233. A new  "--since" argument can be used on the command line when requesting fwding history. It can be used with start_time or end_time or neither. If neither are specified then a negative value is expected, .e.g., -12D which means that the 12 (twelve) previous days of data of requested. This is done by setting start time to present time minus 12 days and end time to present. If start_time is present then a positive value is expected and end_time will be sent to start time plus the "since" value. If end time is specified then a negative value is expected for since and start time will be end time *plus* this negative value. Errors include specifying a positive value for since with only an end time or a negative value for since with only a start time. Currently H, D, or W can be used, respectively for hours, days or weeks.

### CODE REVIEW

**Comment by jrmillerwork** [comment]
Premature.

**Comment by jrmillerwork** [comment]
Fixed merge issue

File: cmd/lncli/commands.go [hunk 1]
```diff
@@ -3173,10 +3174,36 @@ var forwardingHistoryCommand = cli.Command{
...
+		case unit == "H" :
+			ret = int64(3600*count)
+		case unit == "D" :
+		    ret = int64(24*3600*count)
+		case unit == "W" :
+			ret = int64(7*24*3600*count)  
+		default:
+		    err = errors.New("Time unit must be H,D or W")	  	
+	}
+	}
```
Comment (jraedisch) [inline cmd/lncli/commands.go]: You should run gofmt before commiting. It makes worrying about code formatting a non-issue.
There are many tools (e.g. go-plus for Atom), that do this automatically for you.

Comment (jrmillerwork) [reply cmd/lncli/commands.go]: I will do this. Sorry, I am sure that was in the instructions for contributors.

File: cmd/lncli/commands.go [hunk 2]
```diff
@@ -3173,10 +3174,36 @@ var forwardingHistoryCommand = cli.Command{
 			Name:  "max_events",
 			Usage: "the max number of events to return",
 		},
+		cli.StringFlag{
+			Name:  "since",
+			Usage: "number of H(ours)/D(ays)/W(eeks) before end_time (if negative) or after start_time or current time if no start_time specifued",
```
Comment (jraedisch) [inline cmd/lncli/commands.go]: This seems a bit too much functionality for one param.
I started giving it some thought and will probably add some proposal later.

In any case I think letters should be lower case, and there is a small typo in the last word: `specifued`

Comment (jrmillerwork) [reply cmd/lncli/commands.go]: I think if the param is called Interval and it is always positive than start_time and Interval results in end_time = start_time plus Interval; if end_time then start_time = end_time - interval. If only Interval, then end_time = present and start_time = present - Interval. How's that?

Comment (jrmillerwork) [reply cmd/lncli/commands.go]: Month is not one of the choices; I think Week is enough. Not sure what the problem with the rolling window is: just saying --interval 3W with no start or end time I think gives us that, in this case start time is three weeks from present and end time is present. Am I missing something?

**Comment by jrmillerwork** [comment]
Not ready for review.

**Pull request closed without merging**